cmd/carbon-relay-ng/carbon-relay-ng_test.go

```diff
@@ -339,3 +340,41 @@ func BenchmarkTableDisPatchAndEndpointReceive(b *testing.B) {
 	}
 	tE.Close()
 }
+
+func TestConfigEnvVarInterpolation(t *testing.T) {
+	os.Setenv("GRAFANA_NET_ADDR", "foo.com")
+	os.Setenv("GRAFANA_NET_API_KEY", "wow")
+	os.Setenv("GRAFANA_NET_USER_ID", "123")
+	os.Setenv("HOST", "foo.bar.baz")
+
+	defer os.Unsetenv("GRAFANA_NET_ADDR")
+	defer os.Unsetenv("GRAFANA_NET_API_KEY")
+	defer os.Unsetenv("GRAFANA_NET_USER_ID")
+	defer os.Unsetenv("HOST")
+
+
+	template := []byte(`
+host = "${HOST}"
+# [[route]]
+key = 'grafanaNet'
+type = 'grafanaNet'
+addr = "${GRAFANA_NET_ADDR}"
+apikey = "${GRAFANA_NET_USER_ID}:${GRAFANA_NET_API_KEY}"
+`)
+
+	expected_template := `
+host = "work"
```

Comment:
This test fails for me locally, I'm guessing `work` is your hostname :)

We should probably drop the `HOST` env var since it's not actually used.

I'm a little torn on if we should remove `host` from this test altogether or substitute in the host name dynamically.

Reply:
Ha, good catch - I left the support for the HOST variable as it was there originally - I can of course update the test to get the value from the current environment to make it reproducible.

Reply:
@fitzoh Updated!

Reply:
Awesome!
I pushed a small tweak in 1070d4389adb4b1b8b63a47acfe0d73ee2bd5071 to split the hostname + env var interpolation into separate tests and accounted for the way it splits on `.`

LGTM, I'll poke the team and see if I can get another quick +1